### PR TITLE
Fix #10190 - Zero DistrictHeatingStream reported in ABUPS

### LINKED ISSUE #10190: Zero DistrictHeatingSteam for ExteriorLightsAndEq.idf in eplusout.sql
Issue overview
--------------

The idf contains:
```
  Exterior:FuelEquipment,
    OutsideSteamEq,          !- Name
    DistrictHeatingSteam,    !- Fuel Use Type
    ON,                      !- Schedule Name
    1000;                    !- Design Level {W}
```

`eplustbl.htm`: Table "Annual and Peak Values - Other" shows, e.g., "ExteriorEquipment:DistrictHeatingSteam=31.54".

`eplusout.sql`: Table "TabularDataWithStrings", shows:
image

I'd expect "RowName=Exterior Equipment" to be non-zero. (District Heating Water appears to correctly have 31.54.)

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added: https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/10000s/10190/mcve.idf
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #10190 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [x] If structural output changes, add to output rules file and add OutputChange label
 - [x] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by dareumnam** [comment]
Thanks so much for the quick fix! @jmarrec

File: src/EnergyPlus/OutputReportTabular.cc [hunk 1]
```diff
@@ -1751,7 +1751,7 @@ void GetInputOutputTableSummaryReports(EnergyPlusData &state)
         ort->resourceTypeNames(2) = "NaturalGas";
         ort->resourceTypeNames(3) = "DistrictCooling";
         ort->resourceTypeNames(4) = "DistrictHeatingWater";
-        ort->resourceTypeNames(5) = "DistirctHeatingSteam";
+        ort->resourceTypeNames(5) = "DistrictHeatingSteam";
```
Comment (mjwitte) [inline src/EnergyPlus/OutputReportTabular.cc:1754]: It would be nice to have these draw from `DataGlobals::eResourceNames `.

Comment (jmarrec) [reply src/EnergyPlus/OutputReportTabular.cc:1754]: Done

File: src/EnergyPlus/OutputReportTabular.cc [hunk 2]
```diff
@@ -1747,20 +1747,20 @@ void GetInputOutputTableSummaryReports(EnergyPlusData &state)
...
-        ort->resourceTypeNames(10) = "FuelOilNo1";
-        ort->resourceTypeNames(11) = "FuelOilNo2";
-        ort->resourceTypeNames(12) = "Propane";
-        ort->resourceTypeNames(13) = "OtherFuel1";
-        ort->resourceTypeNames(14) = "OtherFuel2";
+        ort->resourceTypeNames(1) = Constant::eResourceNames[static_cast<int>(Constant::eResource::Electricity)];
+        ort->resourceTypeNames(2) = Constant::eResourceNames[static_cast<int>(Constant::eResource::NaturalGas)];
+        ort->resourceTypeNames(3) = Constant::eResourceNames[static_cast<int>(Constant::eResource::DistrictCooling)];
+        ort->resourceTypeNames(4) = Constant::eResourceNames[static_cast<int>(Constant::eResource::DistrictHeatingWater)];
+        ort->resourceTypeNames(5) = Constant::eResourceNames[static_cast<int>(Constant::eResource::DistrictHeatingSteam)];
```
Comment (jmarrec) [inline src/EnergyPlus/OutputReportTabular.cc:1754]: Rely on eResourceNames, to avoid typos

File: tst/EnergyPlus/unit/OutputReportTabular.unit.cc [hunk 3]
```diff
@@ -12586,3 +12586,174 @@ TEST_F(SQLiteFixture, UpdateSizing_EndSysSizingCalc)
     Real64 return_val = execAndReturnFirstDouble(query);
     EXPECT_EQ(return_val, 5080.22);
 }
+
+TEST_F(SQLiteFixture, OutputReportTabular_DistrictHeating)
+{
+    // Test for #10190 - District Heating Steam is empty
+    state->dataSQLiteProcedures->sqlite->createSQLiteSimulationsRecord(1, "EnergyPlus Version", "Current Time");
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/OutputReportTabular.unit.cc:12593]: Unit test

**Comment by jmarrec** [comment]
With MCVE on dev support:

## Before fix

image


image

## After fix

image


image

File: src/EnergyPlus/OutputReportTabular.cc (hunk 2 above)
Comment (Myoldmopar) [reply src/EnergyPlus/OutputReportTabular.cc:1754]: I love that.

**Review by Myoldmopar — APPROVED** [review]
This is obviously a good step forward.  It was a clear typo before, and now that we are relying on the built-in names, it will be much harder for a typo to sneak in.  I do wonder if you intentionally put a typo in the PR name though, or if that's simply a bit of irony?

**Comment by Myoldmopar** [comment]
The MCVE is a bit funny that the steam and water energy are the same, and they add up?  Is something being double counted?  I may just be misunderstanding the MCVE.  Anyway, I am not blaming anything on this PR, it was obviously a typo before and is now fixed.  CI is completely happy, no issues or diffs.  I'm going to test this out and merge if ready with develop pulled in.

Tahnks @jmarrec !

**Comment by Myoldmopar** [comment]
Yep, everything is happy.  If this still needs anything else, we can address it later, but I'm merging this now.  Thanks @jmarrec

**Comment by jmarrec** [comment]
The MCVE uses two objects, one steam one water, with the same inputs.

**Comment by Myoldmopar** [comment]
:+1:

**Pull request merged**